src/inverted-index.js

```diff
@@ -1,49 +1,86 @@
+  this.stopWords = ["a", "an", "and", "as", "at", "but", "by", "each", "every", "for",
+    "from", "her", "his", "in", "into", "its", "like", "my", "no", "nor",
+    "of", "off", "on", "onto", "or", "our", "out", "outside", "over", "past",
+    "since", "so", "some", "than", "that", "the", "their", "this", "to", "up", "with"
   ];
 
-  this.fileContents = [];
-  this.index_ = {};
-  // this.createIndex = function(filepath) {
-  //   fetch(filepath)
-  //     .then(function(response) {
-  //       response.json();
-  //       return response.json();
-  //     });
-  // };
+  this.readFile = function (filepath) {
+    var self = this;
+    return fetch(filepath)
+      .then(function (response) {
+        return response.text();
+      }).then(function (response) {
+        try {
+          self.fileContents = JSON.parse(response);
+        } catch (e) {
+          console.log(e);
+        }
+        return response;
+      }).catch(function (err) {
+        console.log('parsing failed', err);
+        throw err;
+      });
+  };
+
+  this.getIndex = function () {
+    var self = this;
 
-  this.getIndex = function() {
-     var self = this;
-     for (var i = 0; i < book.length; i++) {
-       var obj = book[i];
-       Object.keys(obj).map(function(key) {
-         self.fileContents.push(obj[key]);
-       });
-       strFileContents = this.fileContents.toString().toLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]+/gi, " ").replace(/\s{2,}/g, " ").trim().split(" ");
-      //  console.log(i);
-       for (var x = 0; x < strFileContents.length; x++) {
-         this.index_[strFileContents[x]] = i;
+    this.fileContents.forEach(function (element, index) {
+      for (var value in element) {
+        var strFileContents = element[value].toLowerCase()
+          .replace(/\W/gi, " ").replace(/\s{2,}/g, " ")
+          .trim().split(" ");
+        strFileContents.forEach(function (word) {
+          // search for and exclude stop words from index creation
+          if (self.stopWords.indexOf(word) === -1) {
+            if (self.invertedIndex[word]) {
+              var currentValue = self.invertedIndex[word];
+              if (currentValue.indexOf(index) === -1) {
+                currentValue.push(index);
+              }
+            } else {
+              self.invertedIndex[word] = [index];
+            }
+          }
+        });
+      }
+    });
+    return this.invertedIndex;
+  };
+
+  this.searchIndex = function (searchTerms) {
+    var self = this;
+    var terms = [];
+    var searchResults = [];
 
-       }
-       console.log(this.index_);
-     }
-    //  console.log(this.index_);
-    //  strFileContents = this.fileContents.toString().toLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]+/gi, " ").replace(/\s{2,}/g, " ").trim().split(" ");
+    // convert all format of input into an array
+    if (!Array.isArray(searchTerms)) {
+      for (var key in arguments) {
+        terms.push(arguments[key]);
+      }
+    } else {
```

Comment:
**Code Quality - Nitpick**

What if a user searches the index like this `index.searchIndex('alice in wonderland')`